x/common/error.go

```diff
@@ -0,0 +1,47 @@
+package common
+
+import (
+	"errors"
+	"fmt"
+)
+
+// ToError converts a value to error if if (1) is a string, (2) has a String() function
+// or (3) is already an error.
+func ToError(v any) error {
+	if v == nil {
+		return nil
+	}
+	switch v := v.(type) {
+	case string:
+		return errors.New(v)
+	case error:
+		return v
+	case fmt.Stringer:
+		return errors.New(v.String())
+	default:
+		panic(fmt.Errorf("invalid type: %T", v))
+	}
+}
+
+// Combines errors into single error. Error descriptions are ordered the same way
+// they're passed to the function.
+func CombineErrors(errs ...error) error {
+	var err error
+	for _, e := range errs {
+		switch {
+		case e != nil && err == nil:
+			err = e
+		case e != nil && err != nil:
+			err = fmt.Errorf("%s: %s", err, e)
```

Comment:
1. I originally had it using `errors.Wrap` like you're suggesting, but I didn't like that when you test it, calling `.Error()` on the test cases made the tests very brittle. 
2. Having to register every error with the SDK to use wrap (supposed to be built-in) feels too restrictive since it comes with the [`fmt` and `errors` packages](https://pkg.go.dev/cosmossdk.io/errors#Wrap) by default.  
    via `fmt.Errorf("%w", err)` and `errors.Wrap`  
    I tried to write these abstractions in such a way that you wouldn't need the sdk dependencies. 
3. Both `TryCatch` and the other functions in common/error.go can be used in any Golang application since they only depend on the standard library.